evalml/pipelines/components/transformers/samplers/undersampler.py

```diff
@@ -0,0 +1,93 @@
+import pandas as pd
+from skopt.space import Integer, Real
+
+from evalml.pipelines.components.transformers import Transformer
+from evalml.preprocessing.data_splitters.balanced_classification_sampler import (
+    BalancedClassificationSampler
+)
+from evalml.utils.woodwork_utils import (
+    _convert_woodwork_types_wrapper,
+    infer_feature_types
+)
+
+
+class Undersampler(Transformer):
+    """Random undersampler component."""
+    name = "Undersampler"
+    hyperparameter_ranges = {
+        'balanced_ratio': Real(1, 10),
+        'min_samples': Integer(50, 1000),
+    }
+
+    def __init__(self, balanced_ratio=4, min_samples=100, min_percentage=0.1, random_seed=0, **kwargs):
+        """Initializes an undersampling transformer to downsample the majority classes in the dataset.
+
+        Arguments:
+            balanced_ratio (float): The largest majority:minority ratio that is accepted as 'balanced'. For instance, a 4:1 ratio would be
+                represented as 4, while a 6:5 ratio is 1.2. Must be greater than or equal to 1 (or 1:1). Defaults to 4.
+            min_samples (int): The minimum number of samples that we must have for any class, pre or post sampling. If a class must be downsampled, it will not be downsampled past this value.
+                To determine severe imbalance, the minority class must occur less often than this and must have a class ratio below min_percentage.
+                Must be greater than 0. Defaults to 100.
```

Comment:
@chukarsten Yep, if used in AutoMLSearch, it is fairly redundant with the `invalidtargetdatacheck` when the data is imbalanced. I think it's nice to leave in if users use these pipelines outside of AutoMLSearch, such as if they pull out pipelines after searching and train it on their local machine. Because of that, I think it's fine to leave this in, as it won't throw errors during AutoMLSearch, but what do you think?

Reply:
Agreed, I think I should have better considered usage outside of automl.  I think from previous reviews that you alert the user to triggering these imbalances so I think this is good to go.